cpp/src/arrow/compute/kernels/scalar_arithmetic.cc

```diff
@@ -852,24 +854,235 @@ struct LogbChecked {
   }
 };
 
+struct RoundUtil {
+  template <typename T>
+  static constexpr enable_if_t<std::is_floating_point<T>::value, bool> IsApproxEqual(
+      const T a, const T b, const T abs_tol = kDefaultAbsoluteTolerance) {
+    return std::fabs(a - b) <= abs_tol;
+  }
```

Comment:
Also, [compare.cc](https://github.com/apache/arrow/blob/master/cpp/src/arrow/compare.cc#L78-L86) contains an approximate equal for floating point values.

Reply:
Since these methods are not exposed, I will keep the ones in `RoundUtils`.